lib/gtl_formatter.rb

```diff
@@ -27,7 +27,8 @@ class GtlFormatter
   ].freeze
 
   COLUMN_WITH_IMAGE = {
-    'ext_management_system.name' => 'fonticon_or_fileicon'
+    'ext_management_system.name' => 'fonticon_or_fileicon',
+    'resource.name'              => 'resource_fonticon_or_fileicon',
```

Comment:
This kind of scares me because resource is the generic name for _any_ polymorphic reference.  Maybe we can add a check that the resource is an ExtManagementSystem type, but I'm not sure since I'm having a hard time following this code.

Perhaps instead we should have an ext_management_system association on the keypair (and the other objects).  @agrare Thoughts here?

Reply:
Is it possible to check the `resource_type` here?  We could also do a virtual_column for the `ManageIQ::Providers::CloudManager::AuthKeyPair`

We could add an `authentications#ems_id` and migrate all of the resource_type/resource_id relations over to that but that is a lot more change relatively speaking.

Reply:
We might be able to check the resource type within the `resource_fonticon_or_fileicon` method and only support it when it's an EMS, but I'm concerned that the check for COLUMN_WITH_IMAGE happens before that method is called (i.e. once it's in this list it will always expect an image)

My thought was to have either a virtual association or perhaps a _real_ association in `ManageIQ::Providers::CloudManager::AuthKeyPair`, but the association uses the resource_type/resource_id and only accepts EMS types in the resource_type.